### ISSUE TITLE
Feature Request - Allow specifying layout slots as array in custom card list settings to keep order

### ISSUE THREAD

**Issue by joshbirnholz** [Senryoku/Draftmancer#783]
I'm writing some code that generates custom card lists files, based on XML files for Cockatrice custom sets. I'm writing this code in Swift. Since Dictionaries in swift do not have order, encoding the layouts to a JSON string gives me inconsistent results, since the object keys are encoded in order based on the internal Swift dictionary, which is different each time it is encoded.

```JSON
{
  "layouts" : {
    "Mythic" : {
      "slots" : {
        "Common" : 10,
        "Land" : 1,
        "Mythic" : 1,
        "Uncommon" : 3
      },
      "weight" : 1
    },
    "Rare" : {
      "slots" : {
        "Uncommon" : 3,
        "Rare" : 1,
        "Common" : 10,
        "Land" : 1
      },
      "weight" : 7
    }
  },
  "withReplacement" : true
}
```

I'd like the slots to be in a specific order, however, when drafting on Draftmancer (ie Land, Rare, Uncommon, Common). If we could specify the slots the slots as an array like this, and have that be respected when drafting, that would be great!

Something like this is what I would want to have be valid:

```JSON
{
  "layouts" : {
    "Mythic" : {
      "slots" : [
        {
          "slot": "Land",
          "quantity": 1
        },
        {
          "slot": "Mythic",
          "quantity": 1
        },
        {
          "slot": "Uncommon",
          "quantity": 3
        },
        {
          "slot": "Common",
          "quantity": 10
        }
      ],
      "weight" : 1
    },
    "Rare" : {
      "slots" : [
        {
          "slot": "Land",
          "quantity": 1
        },
        {
          "slot": "Rare",
          "quantity": 1
        },
        {
          "slot": "Uncommon",
          "quantity": 3
        },
        {
          "slot": "Common",
          "quantity": 10
        }
      ],
      "weight" : 7
    }
  },
  "withReplacement" : true
}
```

**Comment by Senryoku** [comment]
Implemented in a92a240

I went with `name` and `count` to be more consistent with the rest of the settings:
```json
"layouts": {
  "Normal" : {
    "weight": 3,
    "slots": [
      { "name": "Rare",     "count": 1},
      { "name": "Uncommon", "count": 3},
      { "name": "Common",   "count": 10}
    ]
  },
  "Lucky": {
    "weight": 1,
    "slots": [
      { "name": "Rare"     , "count" : 2},
      { "name": "Uncommon" , "count" : 3},
      { "name": "Common"   , "count" : 9}
    ]
  }
}
```

**Comment by joshbirnholz** [comment]
Awesome, thank you!
